CMakeLists.txt

```
@@ -62,6 +62,20 @@ if(JWT_EXTERNAL_PICOJSON)
   find_package(picojson 1.3.0 REQUIRED)
 endif()
 
+find_package(nlohmann_json CONFIG)
+
+if(NOT nlohmann_json_FOUND)
+  include(FetchContent)
+
+  FetchContent_Declare(
+    json
+    GIT_REPOSITORY https://github.com/ArthurSonzogni/nlohmann_json_cmake_fetchcontent.git
```

As much as I understand the benefits, but rather see the real repository. 

https://github.com/Kitware/CMake/blob/8251cdfe39c33cbcc38479f0245da3a858a8152f/Modules/FetchContent.cmake#L123

> This "first to record, wins" approach is what allows hierarchical projects to have parent projects override content details of child projects.

```suggestion
    GIT_REPOSITORY https://github.com/nlohmann/json.git
```

---

Are you sure about this ? Main repository is really huge and it takes time to download all of it.

---

It looks more official using the real one, that's just me.

You only pay the price once when you clone + you can install system wide and avoid it (I need to check the logs first) + downstream projects can override it

***

Don't change it, I think it's good to merge test the performance across the CI workflows can wait for me

---

ok, thanks for the review, finally CI is OK ;)

---

I know this has already been merged, but I think a good alternative would be to pull the official release tar.
It's official, but unlike the git pull its only a couple hundret K, not a couple hundret megs.
E.g.
```cmake
FetchContent_Declare(nlohmann_json
    URL https://github.com/nlohmann/json/releases/download/v3.11.2/json.tar.xz
    DOWNLOAD_EXTRACT_TIMESTAMP OFF
)
FetchContent_MakeAvailable(nlohmann_json)
```